CMakeLists.txt

```diff
@@ -516,7 +516,9 @@ if(LINTER)
 
   if(CLANG_TIDY)
     message(STATUS "Linter support (clang-tidy) enabled")
-    set(CMAKE_C_CLANG_TIDY "${CLANG_TIDY};--quiet")
+    set(CMAKE_C_CLANG_TIDY
+        "${CLANG_TIDY};--warnings-as-errors=clang-diagnostic-*,clang-analyzer-*,-*,clang-analyzer-core.*,clang-diagnostic-*"
```

Comment:
Probably we can write `-warnings-as-errors=*` here. It doesn't enable the checks, and we won't have to maintain the list of checks in two different places.

Reading the docs here: https://clang.llvm.org/extra/clang-tidy/

Reply:
And according to Sven's comment, it might make sense to put this not here, but in the CI build settings, so that the warnings are treated as errors only the in CI. Not sure if people need clang-tidy at all in their default MacOS build settings. I don't, but I use Linux.

Reply:
OK, I will see if we can add it only on CI.

Regarding `-warnings-as-errors=*` I tried it locally and discovered that we have too many warnings in this case. Don't think we have the resources to fix them now, nor am I sure that this is necessary.

Reply:
Oh, I thought we have `-checks=<some closed list>`, no warnings with this list, and then `-warning-as-errors=*` could be enabled. I'd treat the clang-tidy warnings as normal warnings and make sure that there are none, so that the new ones are  easy to notice. A configuration that emits a lot of warnings feels inconvenient to use. But never mind, do as you see fit.

Reply:
The list I used here is a default list of checks (at least as I understood it from the docs). Ideally, I would like to have something like -Werror, but unfortunately clang-tidy doesn't seem to have such a flag. Thus I had to specify the list of checks that should be treated as errors manually 🤷‍♂️